articles/113_ros_time.md

```diff
@@ -0,0 +1,154 @@
+
+Another important use case for using an abstracted time source is when you are running logged data against a simulated robot instead of a real robot.
+Depending on the simulation characteristics, the simulator may be able to run much faster than real time or it may need to run much slower.
+Running faster than real time can be valuable for high level testing as well allowing for repeated system tests.
+Slower than real time simulation is necessary for complicated systems where accuracy is more important than speed.
+Often the simulation is the limiting factor for the system and as such the simulator can be a time source for faster or slower playback.
+Additionally if the simulation is paused the system can also pause using the same mechanism.
+
+### Challenges in using abstracted time
+
+There are many algorithms for synchronization and they can typically achieve accuracies which are better than the latency of the network communications between devices on the network.
+However, these algorithms take advantage of assumptions about the constant and continuous nature of time.
+
+An important aspect of using an abstracted time is to be able to manipulate time.
+In some cases, speeding up, slowing down, or pausing time entirely is important for debugging.
+
+The ability to support pausing time requires that we not assume that the time values are always increasing.
+
+When communicating the changes in time propagation, the latencies in the communication network becomes a challenge.
+Any change in the time abstraction must be communicated to the other nodes in the graph, but will be subject to normal network communication latency.
+This inaccuracy is proportional to the latency of communications and also proportional to the increase in the real time factor.
+If very accurate timestamping is required when using the time abstraction, it can be achieved by slowing down the real time factor such that the communication latency is comparatively small.
+
+The final challenge is that the time abstraction must be able to jump backwards in time, a feature that is useful for log file playback.
+This behavior is similar to a system clock after a negative date change, and requires developers using the time abstraction to make sure their algorithm can deal with the discontinuity.
+Appropriate APIs must be provided for to the developer API to enable notifications of jumps in time, both forward and backwards.
+
+There will be at least three versions of these abstractions with the following types.
+There are several use cases where using the ROS time abstraction are inappropriate.
+
+- SystemTime: which will follow the system clock.
+- SteadyTime: which will follow the steady clock on the system.
+- ROSTime: which will follow the system clock unless it's designated Time Source is active.
+
+### System Time
+
+For convenience in these cases we will also provide the same API as above, but use the name `SystemTime`.
```

Comment:
The layout of the text makes "these cases" sort of ambiguous. I'd suggest describing which cases directly here instead.